### PR TITLE
Improve and fix 2D polygon features and tests.

### PR DESCRIPTION
**Pull request by tibuch**
All polygon features are tested and fixed https://github.com/imagej/imagej-ops/issues/449.

The test epsilon is set to 10e-12.

Default, min and max feret angle/diameter added.

Feret diameter for a given angle added.

Removed expensive tests.

@ctrueden if the test is too expensive I can create a smaller test image.

### CODE REVIEW

File: src/main/java/net/imagej/ops/geom/geom2d/DefaultEccentricity.java [hunk 1]
```diff
@@ -61,10 +62,13 @@ public void initialize() {
 
 	@Override
 	public void compute1(final Polygon input, final DoubleType output) {
-		output.set(majorAxisFunc.compute1(input).getRealDouble() /
-			minorAxisFunc.compute1(input).getRealDouble());
+
+		final double a = majorAxisFunc.compute1(input).get() / 2.0;
```
Comment (dietzc) [inline src/main/java/net/imagej/ops/geom/geom2d/DefaultEccentricity.java]: we could operate only on `DoubleType` here: `majorAxisFunc.compute1(input).get().copy().div(2.0)` etc.

File: src/main/java/net/imagej/ops/geom/geom2d/DefaultElongation.java [hunk 2]
```diff
@@ -30,36 +30,56 @@
...
+		final RealLocalizable p2 = minBB.get(1);
+		final RealLocalizable p3 = minBB.get(2);
+
+		double width = Math.sqrt(Math.pow(p1.getDoublePosition(0) - p2.getDoublePosition(0), 2)
+				+ Math.pow(p1.getDoublePosition(1) - p2.getDoublePosition(1), 2));
+		double length = Math.sqrt(Math.pow(p2.getDoublePosition(0) - p3.getDoublePosition(0), 2)
+				+ Math.pow(p2.getDoublePosition(1) - p3.getDoublePosition(1), 2));
+
+		if (width > length) {
+			double tmp = width;
```
Comment (dietzc) [inline src/main/java/net/imagej/ops/geom/geom2d/DefaultElongation.java]: final

File: src/main/java/net/imagej/ops/geom/geom2d/DefaultFeretsAngle.java [hunk 3]
```diff
@@ -44,50 +41,26 @@
...
+		implements Ops.Geometric.FeretsAngle {
 
-	private UnaryFunctionOp<Polygon, Pair<RealLocalizable, RealLocalizable>> function;
-
-	@SuppressWarnings({ "unchecked", "rawtypes" })
 	@Override
-	public void initialize() {
-		function = (UnaryFunctionOp) Functions.unary(ops(), Ops.Geometric.Feret.class, Pair.class, in());
-	}
+	public void compute1(final Pair<RealLocalizable, RealLocalizable> input, final DoubleType output) {
```
Comment (dietzc) [inline src/main/java/net/imagej/ops/geom/geom2d/DefaultFeretsAngle.java]: can you provide some javadoc which explain your assumptions / formula?

File: src/main/java/net/imagej/ops/geom/geom2d/DefaultMinimumFeret.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+		function = Functions.unary(ops(), Ops.Geometric.ConvexHull.class, Polygon.class, in());
+	}
+
+	@Override
+	public Pair<RealLocalizable, RealLocalizable> compute1(Polygon input) {
+		final List<? extends RealLocalizable> points = function.compute1(input).getVertices();
+
+		double distance = Double.POSITIVE_INFINITY;
+		RealLocalizable p0 = points.get(0);
+		RealLocalizable p1 = points.get(0);
```
Comment (dietzc) [inline src/main/java/net/imagej/ops/geom/geom2d/DefaultMinimumFeret.java]: final

File: src/main/java/net/imagej/ops/geom/geom2d/DefaultEccentricity.java (hunk 1 above)
Comment (tibuch) [inline src/main/java/net/imagej/ops/geom/geom2d/DefaultEccentricity.java]: This only works with DoubleType as argument.

**Comment by tibuch** [comment]
@dietzc code and java-doc is cleaned up.

**Comment by dietzc** [comment]
@haesleinhuepf any comments?

**Comment by tibuch** [comment]
Just renamed some classes. Now all features start with Default-. Amended the changes to the last commit.

**Pull request merged**